File: cs50x/pset6/sentimental-cash/cash.py
from cs50 import get_float

# Ask for change
while True:
    change = get_float("Change: ")
    if change > 0:
        break

# Convert dollars to cents
change = round(change * 100)

# Count
coins = 0

# Assign coins value
quarter = 25
dime = 10
nickel = 5
penny = 1

coins += change // 25  # divide change by quarter (25), add the result to coins
change %= 25         # use mod quarter (25) to calculte the remaining change

coins += change // 10  # repeat the process, divide with 10
change %= 10          # remaining change

coins += change // 5  # repeat the process, divide with 5
change %= 5          # remaining change

coins += change // 1  # no need to use mod because the rest is now 0

print(coins)

File: cs50x/pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int count(int change);

int main(void)
{
    int change;
    do
    {
        change = get_int("Change owed: ");
    }
    while (change < 0);

    int coins = count(change);
    printf("%i\n", coins);
}

int count(int change) // function to count the minimal number of coins required for change
{
    // count
    int coins = 0;

    // assign coins value for better reading
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;

    coins += change / quarter; // divide change by quarter (25), add the result to coins
    change %= quarter;         // use mod quarter (25) to calculte the remaining change

    coins += change / dime; // repeat the process, divide with 10
    change %= dime;         // remaining change

    coins += change / nickel; // repeat the process, divide with 5
    change %= nickel;         // remaining change

    coins += change / penny; // no need to use mod because the rest is now 0

    return coins;
}
